hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/RouterWebServices.java

```diff
@@ -424,7 +424,7 @@ public BulkActivitiesInfo getBulkActivities(
       MediaType.APPLICATION_XML + "; " + JettyUtils.UTF_8 })
   @Override
   public AppActivitiesInfo getAppActivities(@Context HttpServletRequest hsr,
-      @QueryParam(RMWSConsts.APP_ID) String appId,
+      @PathParam(RMWSConsts.APPID) String appId,
```

Comment:
After comparing the code with the ResourceManager (RM), it can be confirmed that there are some issues with this part of the code.

https://github.com/apache/hadoop/blob/e4eda40ac9966a7d8488bbb0630eaa02c3f2b0e0/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/RMWebServices.java#L877C3-L892C31